### DISCUSSION TITLE
need guidance on how to fill the webhid HIDCollectionInfo using hidapi APIs

### DISCUSSION THREAD

**Discussion by Codeskb** [libusb/hidapi#629] [Q&A]
Hello all, I am writing an adapter between webhid and hidapi. I need some guidance how to fill the [HIDCollectionInfo](https://wicg.github.io/webhid/#hidcollectioninfo-dictionary) for a particular device using hidapi APIs. Webhid provides an algorithm (https://wicg.github.io/webhid/#dfn-parsing-the-report-descriptor) how to parse the raw bytes of report descriptor and I see that hidapi provides hid_get_report_descriptor to get the raw data. Did anyone attempt to do this ? Is there any implementation I can take as reference ?
Thanks.

**Comment by Youw** [comment]
Essentially I believe this discussion is not directly related to HIDAPI.
As you already pointed, HIDAPI provides `hid_get_report_descriptor` to get a RAW HID Report Descriptor.

It is probably best to post this somewhere at _webhid_ community (is there a discussion/forum/etc.?): "how to fill the `HIDCollectionInfo` having the RAW HID Report Descriptor".
I have a feeling someone has already done it in the past and you might be able to find the complete implementation to do so.
